internal/storage/gcs/bucket.go

```diff
@@ -20,6 +20,36 @@ import (
 	"golang.org/x/net/context"
 )
 
+// ObjectWriter interface represents asynchronous object
+// writers, which don't write the entire GCS object from
+// content in a single-shot. These objects are created once,
+// are written to multiple times and are eventually closed
+// once. On closing, they return a GCS object (gcs.Object)
+// and and error object for err-handling.
+// TODO: rename this to AsynchronousGcsObjectWriter
+type ObjectWriter interface {
```

Comment:
I would prefer keeping the names consistent with the GCS api instead of creating our own names and definitions.

ObjectWriter -> ChunkUploader
write -> uploadChunkAsync

Reply:
Done.